great_expectations/experimental/datasources/sqlite_datasource.py

```diff
@@ -0,0 +1,121 @@
+
+        Returns:
+            This SqliteTableAsset so we can use this method fluently.
+        """
+        self.column_splitter = SqliteYearMonthSplitter(
+            column_name=column_name,
+        )
+        return self
+
+
+@pydantic_dc.dataclass(frozen=True)
+class SqliteYearMonthSplitter(ColumnSplitter):
+    method_name: Literal["split_on_year_and_month"] = "split_on_year_and_month"
+    param_names: List[Literal["year", "month"]] = pydantic.Field(
+        default_factory=lambda: ["year", "month"]
+    )
+
+    def param_defaults(self, table_asset: TableAsset) -> Dict[str, List]:
+        """Query sqlite database to get the years and months to split over.
+
+        Args:
+            table_asset: A SqliteTableAsset over which we want to split the data.
+        """
+        if not isinstance(table_asset, SqliteTableAsset):
+            raise SQLDatasourceError(
+                "Table asset passed to SqliteYearMonthSplitter is not a SqliteTableAsset. It is "
+                f"{table_asset}"
+            )
+
+        return _query_for_year_and_month(
+            table_asset, self.column_name, _get_sqlite_datetime_range
+        )
+
+
+def _get_sqlite_datetime_range(
+    conn: sqlalchemy.engine.base.Connection, table_name: str, col_name: str
+) -> DatetimeRange:
+    q = f"select STRFTIME('%Y%m%d', min({col_name})), STRFTIME('%Y%m%d', max({col_name})) from {table_name}"
+    min_max_dt = [datetime.strptime(dt, "%Y%m%d") for dt in list(conn.execute(q))[0]]
+    return DatetimeRange(min=min_max_dt[0], max=min_max_dt[1])
+
+
+class SqliteDatasource(SQLDatasource):
+    # class var definitions
+    asset_types: ClassVar[List[Type[DataAsset]]] = [SqliteTableAsset]
+
+    # Subclass instance var overrides
+    # right side of the operator determines the type name
+    # left side enforces the names on instance creation
+    type: Literal["sqlite"] = "sqlite"  # type: ignore[assignment]
+    connection_string: str
+    assets: Dict[str, SqliteTableAsset] = {}  # type: ignore[assignment]
+
+    def add_table_asset(
+        self,
+        name: str,
+        table_name: str,
+        order_by: Optional[BatchSortersDefinition] = None,
+    ) -> SqliteTableAsset:
+        """Adds a sqlite table asset to this sqlite datasource.
+
+        Args:
+            name: The name of this table asset.
+            table_name: The table where the data resides.
+            order_by: A list of BatchSorters or BatchSorter strings.
+
+        Returns:
+            The SqliteTableAsset that is added to the datasource.
+        """
+        asset = SqliteTableAsset(
+            name=name,
+            table_name=table_name,
+            order_by=order_by or [],  # type: ignore[arg-type]  # coerce list[str]
+            # see TableAsset._parse_order_by_sorter()
+        )
+        # TODO (kilo59): custom init for `DataAsset` to accept datasource in constructor?
```

Comment:
This was basically lifted from the sql datasource. I'm not sure if I should keep or delete this comment. 

We may want to use the setter for the `datasource` instead of accessing `_datasource` directly. I didn't do that now because I then wanted to typecheck on assignment which led to defining more methods on each class. I can do this in a followup PR.

Reply:
If the comments on the parent method then I think we could delete them here.